lib/diagnostic.js

```diff
@@ -17,7 +17,30 @@ const fs = require('fs'); // jshint ignore: line
 
 const sumLines = (filename, callback) => {
   /* your response here */
-  callback();
+  const inFile = process.argv[2];
+
+  fs.readFile(inFile, 'utf8', (err, data) => {
+    if (err) {
+      console.error(err.stack);
+      return;
+    }
+
+    let dataArray = data.split('\n');
+    let sum = 0;
+
+    dataArray.forEach((num) => {
+      if (parseInt(num, 10).isNaN()) { // if a line is not a number
+        console.log('oops');       // throw an "error" to console
+      } else {
+        sum += parseInt(num, 10);      // else add it to the sum
+      }
+
+    });
+
+  });
+
+  callback(); // wtf do I do with this?
```

Comment:
Your callback is the function that needs to run after you get either an error or data back from this function. Please refer back to previously lessons to see what we did with them.